### DISCUSSION TITLE
Seperate flame/smoke/fire detection

### DISCUSSION THREAD

**Discussion by Bluscream** [blakeblackshear/frigate#13820] [Ideas]
Would be nice if there was a seperate object detector that is trained to see flames in color and under IR lights

**Comment by ebivan** [comment]
Second that. 
I'am trying to detect the right time to put more logs in the oven. So basically I want to detect when the flame goes out.

**Comment by SeanChengN** [comment]
Seconded, I want to monitor my neighbor's electric car in case it burns up my yard.

**Comment by def1149** [comment]
Something like this along with an ESP32 and some code might be a better solution. It can measure temps -40C to 300C so you could set alert temps high or low using this inexpensive thermal camera 

There is already an ESPHome application if you're using Home Assistant https://github.com/infotronic218/esphome-mlx9060
and other implementations if you Google mlx9060.

No image recognition is required 


https://www.adafruit.com/product/4407

**Comment by fl0wfr** [comment]
Would be nice to get the detection of flames/fire even with color cameras, in case of a car or trash catching fire.

**Comment by calco88** [comment]
I have a standalone python tensorflow model running alongside frigate that is trained on flame/fire images only. It doesn't have any IR night training yet, but I can say so far after tweaking some sunlight-related false positives it's stable. I recently had a burn pile on my property and it was going off all day. 

I captured a few images of the burn pile at night, it just looks like a giant white glow. Would be ideal to find a dataset of similar images, because I guarantee my current model will fail at night under IR lights.

**Comment by deanfourie1** [comment]
Any movement on this? This would be a really useful feature.
